tests/ui/methods.rs

```diff
@@ -171,8 +171,27 @@ fn search_is_some() {
                                    x < 0
                                }
                    ).is_some();
-
-    // Check that we don't lint if the caller is not an `Iterator`.
+    
+    let s1 = String::from("hello world");
+    let s2 = String::from("world");
+    // Check caller `find()` is a &`static str case
+    let _ = "hello world".find("world").is_some();
+    let _ = "hello world".find(&s2).is_some();
+    let _ = "hello world".find(&s2[2..]).is_some();
+    // Check caller of `find()` is a String case
+    let _ = s1.find("world").is_some();
+    let _ = s1.find(&s2).is_some();
+    let _ = s1.find(&s2[2..]).is_some();
+    //  Check caller of `find()` is a slice of String case
+    let _ = s1[2..].find("world").is_some();
+    let _ = s1[2..].find(&s2).is_some();
+    let _ = s1[2..].find(&s2[2..]).is_some();
+
+    // Check that we don't lint if `find()` is called with
+    // Pattern that is not a string
+    let _ = s1.find(|c: char| c == 'o' || c == 'l').is_some();
+
+    // Check that we don't lint if the caller is not an `Iterator` or string
     let foo = IteratorFalsePositives { foo: 0 };
     let _ = foo.find().is_some();
     let _ = foo.position().is_some();
```

Comment:
It would be great if you could move those tests to a new file, named after this lint. After that you should be able to add `// run-rustfix` to the test file and simultaneously check if the suggestion is correct.

The `find().is_some()` part of the lint has cases where no suggestion is emitted. You can deal with this by splitting up the files further into `<lint_name>_fixable.rs` and `<lint_name>.rs`.

Reply:
Ok, thanks I will get started on that.

Reply:
Sorry, I should have clarified this earlier, but should I move only the new tests I added (the ones dealing with strings) to the separate file, or all the tests in `search_is_some()`?
https://github.com/rust-lang/rust-clippy/blob/3bd98895f13112912df3a7f23923470b07d53a09/tests/ui/methods.rs#L141

Reply:
All the tests for the `search_is_some` lint please. GH doesn't let you comment on unchanged code ;)

Reply:
Thanks!